test/python/circuit/test_library.py

```diff
@@ -313,6 +314,39 @@ def test_twoq_equivalence(self):
         self.assertTrue(expected.equiv(simulated))
 
 
+@ddt
+class TestGlobalRLibrary(QiskitTestCase):
+    """Test library of global R gates."""
+
+    def test_GR_equivalence(self):
+        """Test global R gate is same as 3 individual R gates."""
+        circuit = GR(num_qubits=3, theta=np.pi/3, phi=2*np.pi/3)
+        expected = QuantumCircuit(3, name="gr")
+        for i in range(3):
+            expected.append(RGate(theta=np.pi/3, phi=2*np.pi/3), [i])
+        self.assertEqual(Operator(expected), Operator(circuit))
```

Comment:
If you test the equality of the circuit by wrapping it into an `Operator` you compute the matrices of the circuit and compare those. It seems to me that you here want to compare the actual gates in the circuit, is that correct?
In that case you just do that by leaving the `Operator` out
```python
self.assertEqual(expected, circuit.decompose())
```
(Same comment for the other tests)